rust-src/ed25519_hd_key_derivation/src/lib.rs

```diff
@@ -0,0 +1,274 @@
+use hmac::{Hmac, Mac};
+use regex::Regex;
+use sha2::Sha512;
+
+const ED25519_CURVE: &[u8; 12] = b"ed25519 seed";
+const HARDENED_OFFSET: u32 = 0x80000000;
```

Comment:
If I understood correctly in the actual derivations for the ConcordiumHdWallet below, this offset limits the range of allowed indices to < 2^31. Is that correct?

Reply:
Yes.

Reply:
That is correct. I have updated the validation to ensure a failure occurs if an index is greater than `2^31`.